dvc/compare.py

```diff
@@ -172,11 +172,17 @@ def as_dict(self, cols: Iterable[str] = None) -> Iterable[Dict[str, str]]:
         ]
 
 
-def _format_field(val: Any, precision: int = None) -> str:
+def _normalize_float(val: float, precision: int):
+    return f"{val:.{precision}g}"
+
+
+def _format_field(
+    val: Any, precision: int = None, round_digits: bool = False
+) -> str:
     def _format(_val):
         if isinstance(_val, float) and precision:
-            fmt = f"{{:.{precision}g}}"
-            return fmt.format(_val)
+            func = round if round_digits else _normalize_float
```

Comment:
I didn't go for  a `f` specifier for rounding because a) we used to use `round` before and, b) `f` is a fixed width formatting, so there could be trailing zeros which `round` does not have.